src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -398,20 +398,11 @@ public synchronized void close() {
                         } catch (SQLException ex) {
                             logger.debug(String.format("Failed to flush before dropping entity '%s'", me.getValue().getEntity().getName()), ex);
                         } finally {
-                            if (insert != null) {
-                                try {
-                                    insert.close();
-                                } catch (Exception e) {
-                                    logger.trace("Could not close prepared statement.", e);
-                                }
-                            }
 
-                            if (insertReturning != null) {
-                                try {
-                                    insertReturning.close();
-                                } catch (Exception e) {
-                                    logger.trace("Could not close prepared statement.", e);
-                                }
+                            try {
+                                me.getValue().close();
```

Comment:
This is only closing the insert statements when the  `properties.isSchemaPolicyCreateDrop()` property is set, which we agreed was strange. 
From our understanding the entity and its statements should be closed regardless, so this if should do an entity close and drop, and an else should do the close. Makes sense?

Reply:
Addressed in 7ade239